api/output_policy.go

```diff
@@ -0,0 +1,197 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+
+	retryablehttp "github.com/hashicorp/go-retryablehttp"
+	"github.com/hashicorp/vault/sdk/helper/jsonutil"
+)
+
+const (
+	ErrOutputPolicyRequest = "output policy request"
+)
+
+var LastOutputPolicyError *OutputPolicyError
+
+type OutputPolicyError struct {
+	*retryablehttp.Request
+	VaultClient     *Client
+	parsingError    error
+	parsedHCLString string
+}
+
+func (d *OutputPolicyError) Error() string {
+	if d.parsedHCLString == "" {
+		d.parseRequest()
+		if d.parsingError != nil {
+			return d.parsingError.Error()
+		}
+	}
+
+	return ErrOutputPolicyRequest
+}
+
+// Builds a sample policy document from the request
+func (d *OutputPolicyError) parseRequest() {
+	capabilities := []string{}
+	switch d.Request.Method {
+	case "GET":
+		capabilities = append(capabilities, "read")
+	case "LIST":
+		capabilities = append(capabilities, "list")
+	case "POST", "PUT":
+		capabilities = append(capabilities, "create")
+		capabilities = append(capabilities, "update")
```

Comment:
Are there operations where we restrict creation but allow updating?

Reply:
This was a good question that had me investigating for awhile but as discussed offline I think we're good here. As mentioned in the [docs](https://www.vaultproject.io/docs/concepts/policies#capabilities), "Very few parts of Vault distinguish between create and update, so most operations require both create and update capabilities. Parts of Vault that provide such a distinction are noted in documentation.".

I locally modified the OpenAPI spec generation code to show endpoints that support Create and Update respectively, and while there is a substantial number of them that claim to support only Update operations, there isn't a great way for us to programmatically verify this at this point in the code, so worst case scenario I think we will just produce some occasional unused "create" capabilities. 

The backend itself should be rejecting operations that don't actually have an implementation, in any case.